daemon/algod/api/server/v2/dryrun.go

```diff
@@ -287,8 +289,17 @@ func (dl *dryrunLedger) Get(addr basics.Address, withPendingRewards bool) (basic
 	if ok {
 		any = true
 		app := dl.dr.Apps[appi]
-		out.AppParams = make(map[basics.AppIndex]basics.AppParams)
-		out.AppParams[basics.AppIndex(app.Id)] = ApplicationParamsToAppParams(&app.Params)
+		params := ApplicationParamsToAppParams(&app.Params)
+		if out.AppParams == nil {
+			out.AppParams = make(map[basics.AppIndex]basics.AppParams)
+			out.AppParams[basics.AppIndex(app.Id)] = params
+		} else {
+			ap, ok := out.AppParams[basics.AppIndex(app.Id)]
+			if ok {
+				MergeAppParams(&ap, &params)
+				out.AppParams[basics.AppIndex(app.Id)] = ap
+			}
```

Comment:
it looks like the symmetrically case of !ok is missing. I can't tell if it's not needed or it's a genuine bug.

Reply:
Good catch, fixed